newsroom/auth_server/models.py

```diff
@@ -39,17 +40,16 @@ def get_allowed_scope(self, scope):
         return ""
 
 
-def query_client(client_id):
-    clients_service = superdesk.get_resource_service("oauth_clients")
+def query_client(client_id: str):
     try:
-        client_data = clients_service.find_one(req=None, _id=ObjectId(client_id))
+        client_data = run_async_to_sync(ClientService().find_by_id(ObjectId(client_id)))
```

Comment:
I would prefer us not to use `run_async_to_sync`, in fact I would like to remove it entirely. As it can cause async/session issues (mainly with database transactions).

If you're unable to use an async function, we have access to the underlying mongo interface (for sync transactions).
Here you could do the following instead:
```python
client_data = ClientService().mongo.find_one({"_id": ObjectId(client_id)})
```

Reply:
got it